frontend/src/components/ViewActivities/activityfns.js

```diff
@@ -0,0 +1,75 @@
+import * as DBFIELDS from '../../constants/database.js';
+import app from '../Firebase';
+
+const db = app.firestore();
+
+/**
+ * Gets the list of activities from the server. 
+ * @param {string} tripId The trip ID.
+ */
+export async function getActivityList(tripId) {
+  return new Promise(function(resolve, reject) {
+    let tripActivities = [];
+    
+    db.collection(DBFIELDS.COLLECTION_TRIPS).doc(tripId)
+        .collection(DBFIELDS.COLLECTION_ACTIVITIES).get()
+        .then(querySnapshot => {
+          querySnapshot.forEach(doc => {
+        let data = doc.data();
+        data['id'] = doc.id;
```

Comment:
What is the purpose of creating a new field with the id (I'm pretty sure that is what this is doing)? I don't see it being used elsewhere.

Reply:
Thinking ahead this is for keeping track of each activity's id for editing purposes.

Reply:
More relevant in my next PR.